pkg/distro/defs/loader.go

```diff
@@ -32,12 +35,140 @@ var (
 	ErrNoPartitionTableForArch    = errors.New("no partition table for arch")
 )
 
-//go:embed */*.yaml
+//go:embed *.yaml */*.yaml
 var data embed.FS
 
-var DataFS fs.FS = data
+var defaultDataFS fs.FS = data
 
-type toplevelYAML struct {
+// distrosYAML defines all supported YAML based distributions
+type distrosYAML struct {
+	Distros []DistroYAML
+}
+
+func dataFS() fs.FS {
+	// XXX: this is a short term measure, pass a set of
+	// searchPaths down the stack instead
+	var dataFS fs.FS = defaultDataFS
+	if overrideDir := experimentalflags.String("yamldir"); overrideDir != "" {
+		olog.Printf("WARNING: using experimental override dir %q", overrideDir)
+		dataFS = os.DirFS(overrideDir)
+	}
+	return dataFS
+}
+
+type DistroYAML struct {
+	// Match can be used to match multiple versions via a
+	// fnmatch/glob style expression. We could also use a
+	// regex and do something like:
+	//   rhel-(?P<major>[0-9]+)\.(?P<minor>[0-9]+)
+	// if we need to be more precise in the future, but for
+	// now every match will be split into "$distroname-$major.$minor"
+	// (with minor being optional)
+	Match string `yaml:"match"`
+
+	// The distro metadata, can contain go text template strings
```

Comment:
I think this comment makes more sense on `executeTemplates` where it is actually implemented. And maybe a bit more documentation about the yaml files being parsed as go templates.